### PR TITLE
Issue238 ice minima

### LINKED ISSUE #238: Break out the minimum ice thresholds
Storing the ice thresholds in IceGrowth means that referencing these values results in pulling in everything in the IceGrowth header, which is a lot of things. In the main executable this doesn't matter, but ice thermodynamics tests can be a bit lighter and still reference the minimum ice thresholds.

Break these values out into a new class which only provide read access to the values, except for IceGrowth, which is a `friend`.

### PR DESCRIPTION
**Pull request by timspainNERSC**
A small part of the TOPAZ ERA branch as a separate PR.

Closes #238

> 
> Storing the ice thresholds in IceGrowth means that referencing these values results in pulling in everything in the IceGrowth header, which is a lot of things. In the main executable this doesn't matter, but ice thermodynamics tests can be a bit lighter and still reference the minimum ice thresholds.
> 
> Break these values out into a new class which only provide read access to the values, except for IceGrowth, which is a friend.

### CODE REVIEW

**Comment by timspainNERSC** [comment]
Includes all of the changes from PR #278

**Comment by timspainNERSC** [comment]
Adds an class containing the minimum ice thickness and concentration values. The class can be considered as part of the `IceGrowth` class, broken out so that other classes can access these values without importing the whole of the `IceGrowth` class.

The class is configured through `IceGrowth` class, rather than its own configuration.

The change in ice concentration is moved to IceGrowth, rather than the `LateralIceSpread` implementation.

The conflicting `MinimumIce` class is removed.

File: physics/test/SlabOcean_test.cpp
[the remark this answers is not in the document]
Comment (timspainNERSC) [reply physics/test/SlabOcean_test.cpp:1]: Rebased

File: physics/src/include/SlabOcean.hpp
[the remark this answers is not in the document]
Comment (timspainNERSC) [reply physics/src/include/SlabOcean.hpp:1]: Rebased

File: physics/src/IceGrowth.cpp [hunk 1]
```diff
@@ -181,24 +201,25 @@ void IceGrowth::lateralIceSpread(size_t i, const TimestepTime& tstep)
         // Note that the cell-averaged hice0 is converted to a ice averaged value
         iLateral->melt(tstep, hice0[i], hsnow[i], deltaHi[i], cice[i], qow[i], deltaCMelt[i]);
     }
-    double deltaC = deltaCFreeze[i] + deltaCMelt[i];
-    cice[i] += deltaC;
-    if (cice[i] >= MinimumIce::concentration()) {
+    deltaCIce[i] = deltaCFreeze[i] + deltaCMelt[i];
```
[the remark this answers is not in the document]
Comment (timspainNERSC) [reply physics/src/IceGrowth.cpp:204]: Indeed, `deltaCIce` will be available as a diagnostic field, but became included here due to my difficulties in disentangling all the changes made as part of the larger TOPAZ/ERA branch.

File: core/test/CMakeLists.txt [hunk 2]
```diff
@@ -185,6 +185,8 @@ add_executable(testPrognosticData
     "${CoreSrc}/${ModelArrayStructure}/ModelArrayDetails.cpp"
     "${CoreSrc}/NZLevels.cpp"
     "${PhysSourceDir}/IceGrowth.cpp"
+    "${PhysSourceDir}/IceMinima.cpp"
```
[the remark this answers is not in the document]
Comment (timspainNERSC) [reply core/test/CMakeLists.txt:188]: But `IceGrowth` is necessary for this test. It's right there in line 187.

File: physics/src/SlabOcean.cpp
[the remark this answers is not in the document]
Comment (timspainNERSC) [reply physics/src/SlabOcean.cpp:1]: Rebased

File: physics/src/modules/include/BasicIceOceanHeatFlux.hpp
[the remark this answers is not in the document]
Comment (timspainNERSC) [reply physics/src/modules/include/BasicIceOceanHeatFlux.hpp:1]: Seems to have been rendered moot by the slab ocean rebase.

File: physics/src/IceGrowth.cpp [hunk 3]
```diff
@@ -181,24 +201,25 @@ void IceGrowth::lateralIceSpread(size_t i, const TimestepTime& tstep)
...
+            updateThickness(hsnow[i], cice[i], deltaCIce[i], 0);
         }
     }
 }
 
 void IceGrowth::applyLimits(size_t i, const TimestepTime& tstep)
 {
-    if (cice[i] < MinimumIce::concentration() || hice[i] < MinimumIce::thickness()) {
+    if ((0. < cice[i] && cice[i] < IceMinima::cMin)
+        || (0. < hice[i] && hice[i] < IceMinima::hMin)) {
```
[the remark this answers is not in the document]
Comment (timspainNERSC) [reply physics/src/IceGrowth.cpp]: Oh, probably. But I'm moving and inlining those functions as a result.

Changed.

**Comment by draenog** [comment]
Personally I would prefer IceMinima to have its own configure() method to set its cMin and hMin members. But it is not a strong preference, so I am happy with merging this PR to develop.

**Review by draenog — APPROVED** [review]

**Pull request merged**